Lemons/main.cpp

```diff
@@ -0,0 +1,97 @@
+//
+//  main.cpp
+//  lemonade
+//
+//  Created by Herman Sollerman on 27/03/16.
+//  Copyright © 2016 Herman Sollerman. All rights reserved.
+//
+
+#include <iostream>
+using namespace std;
+
+int main()
+{
+    int number_of_cases, number_of_days, lemon_per_cup, sugar_per_cup, cups_sold, lemon_price, total_cost=0, sugar_price, j, k;
+    FILE *input;
+
+    input = stdin;
+    fscanf(input, "%d", &number_of_cases);
+    
+    for (k=1; k <= number_of_cases; k++)
```

Comment:
Jag körde väldigt länge med index som börjar på 1, jag tycker det är mest intuitivt. Men jag har också gått ifrån det sedan länge :) Dels för att 99.99% av C/C++ programmerare kör med index som börjar på noll, och dels därför att arrayerna ju börjar med index på noll ... det blir mycket enklare att index också börjar på noll då, så slipper man -1 en massa.

En standard for-loop bör se ut så här i det här fallet:

for (int k = 0; k < number_of_cases; ++k) {
}
Notera att k deklareras i for-loopen (så nära som möjligt där man använder det). Enda undantaget från den regeln är om man måste använda k efter (eller möjligtvis före) själva loopen.
Notera också att jag skriver ++k och inte k++. Det är verkligen en petitess. I vissa fall kan man göra komplicerade objekt/iteratorer där kopiering är en relativt kostsam operation, och om man skriver k++ så kommer kompilatorn först att evaluera k och kopiera undan resultatet, för att sedan räkna upp k självt. skriver man ++k så räknas k upp och resultatet finns då tillgängligt i "original-k" och ingen kopia behövs. Som sagt: petitess ...